src/lib/core/data-table/data-table.ts

```diff
@@ -0,0 +1,323 @@
+})
+export class CdkRowCell {
+  constructor(private columnDef: CdkColumnDef,
+              private elementRef: ElementRef,
+              private renderer: Renderer) {
+    this.renderer.setElementClass(elementRef.nativeElement, `cdk-column-${columnDef.name}`, true);
+  }
+}
+
+/**
+ * Outlet used by the header and data rows that has the right column cells and holds the context.
+ */
+@Directive({selector: '[cdkCellOutlet]'})
+export class CdkCellOutlet {
+  /** The ordered list of cells to render within this outlet's view container */
+  cells: CdkRowCellDef[];
+
+  /** The row data context to be provided to each cell */
+  context: any;
+
+  static mostRecentCellOutlet: CdkCellOutlet = null;
+
+  /**
+   * Sets the most recently constructed CdkCellOutlet with the provided set of cells and context.
+   */
+  static setMostRecentCellOutletData(cells: CdkRowCellDef[], context: any = {}) {
+    CdkCellOutlet.mostRecentCellOutlet.cells = cells;
+    CdkCellOutlet.mostRecentCellOutlet.context = context;
+  }
+
+  constructor(private _viewContainer: ViewContainerRef) {
+    CdkCellOutlet.mostRecentCellOutlet = this;
+  }
+
+  ngOnInit() {
+    this.cells.forEach(cell => {
+      this._viewContainer.createEmbeddedView(cell.template, this.context);
+    });
+  }
+}
+
+/**
+ * Provides a handle for the table to grab the view container's ng-container to insert data rows.
+ */
+@Directive({selector: '[cdkRowPlaceholder]'})
+export class CdkRowPlaceholder {
+  constructor(public viewContainer: ViewContainerRef) { }
+}
+
+/**
+ * Provides a handle for the table to grab the view container's ng-container to insert the header.
+ */
+@Directive({selector: '[cdkHeaderRowPlaceholder]'})
+export class CdkHeaderRowPlaceholder {
+  constructor(public viewContainer: ViewContainerRef) { }
+}
+
+/**
+ * A data table that connects with a data source to retrieve data and renders
+ * a header row and data rows. Updates the rows when new data is provided by the data source.
+ */
+@Component({
+  selector: 'cdk-table',
+  template: `
+    <ng-container cdkHeaderRowPlaceholder></ng-container>
+    <ng-container cdkRowPlaceholder></ng-container>
+  `,
+  host: {
+    'class': 'cdk-table',
+    'role': 'grid' // TODO: Allow the user to choose either grid or treegrid
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CdkTable {
+  /**
+   * Provides a stream containing the latest data array to render. Influenced by the table's
+   * stream of view window (what rows are currently on screen).
+   */
+  @Input() dataSource: DataSource<any>;
+
+  /**
+   * Stream containing the latest information on what rows are being displayed on screen. Offered
+   * to the data source as a heuristic of what data should be provided.
+   */
+  // TODO: Remove max value as the end index and instead calculate the view on init and scroll.
+  viewChange = new BehaviorSubject<CollectionViewer>({start: 0, end: Number.MAX_VALUE});
```

Comment:
`viewChange` -> `viewChanges`? 

(not super into the name, but I don't have an alternative)